src/engine/ruleset/wazuh-core/decoders/core-ossec-message.yml

```diff
@@ -0,0 +1,44 @@
+name: decoder/core-ossec-message/0
+# module: wazuh
+
+metadata:
+  title: OSSEC message decoder
+  description: >
+    Base decoder to process OSSEC message format, parses location part and enriches the events
+    that comes from a Wazuh agent with the host information.
+  compatibility: All wazuh events.
+  versions:
+    - Wazuh 4.*
+  author:
+    name: Wazuh, Inc.
+    date: 06/03/2023
+  references:
+    - https://documentation.wazuh.com/current/development/message-format.html
+    - https://github.com/wazuh/wazuh/issues/15500
+
+definitions:
+  full_location: "[<agent.id>] \\\\(<agent.name>\\\\) <~registered_ip>-><~origin>"
+
+normalize:
+  #### Full location ####
+  #### Present when the event is procesed by remoted
+  - logpar:
+      - wazuh.location: $full_location
+    map:
+      - wazuh.origin: +rename/$~origin
+      - wazuh.registered_ip: +rename/$~registered_ip
+
+  - map:
+      - event.original: $wazuh.message
+      - wazuh.decoders: +array_append/core-ossec-message
```

Comment:
It would be nice to keep the decoder's track at the same point on every decoder, either at the beginning of the "normalize" or at the end of it.